KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,652 @@
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Klaytn platforms (klaytn). -->
+
+The proposed smart contract will be implemented in Solidity and will be compatible with the Ethereum Virtual Machine (EVM). The smart contract will use Ownable contract to restrict access to certain functions to the owner of the contract.
+
+The smart contract will have the following features:
+
+- Add/Remove fund addresses
+- Approve fund addresses and fund allocation
+- Reset the storage values at any unforeseen circumstances before Finalized
+- Finalize the smart contract after execution
+
+### Smart Contracts Overview
+
+#### Enums
+
+The smart contract will have the following enum to track the status of the contract:
+
+- `Initialized - 0`: The initial state of the contract.
+- `Registered - 1`: Senders and receivers registered.
+- `Approved - 2`: Senders approved.
+- `Finalized - 3`: Rebalance executed and finalized.
+
+#### Life Cycle
+
+The contract status should follow the ENUM order above during status change. The only way to go to previous state is by calling Reset() function.
+
+![](../assets/kip-103/lifecycle.png)
+
+Status transition
+
+- Initialized → Registered → Approved → Finalized ✅
+- Initialized → Registered → Finalized ❌
+- Initialized → Approved ❌
+- Initialized → Finalized ❌
+
+#### Structs
+
+The smart contract will have the following structs :
+
+- `Retired`: to store the details of retired and approver addresses.
+- `Newbie`: to store newbie and amount.
+
+#### Storage
+
+The smart contract will have the following storage variables:
+
+- `retirees`: array of retirees structs.
+- `newbie`: array of newbies structs.
+- `status`: current status of the contract.
+- `rebalanceBlockNumber`: Block number of the execution of rebalancing.
+- `memo`: result of the treasury fund rebalance.
+
+#### Modifiers
+
+The smart contract will have the following modifier:
+
+- `atStatus`: to restrict access to certain functions based on the current status of the contract.
+
+#### Constructor
+
+The smart contract will have the following constructor:
+
+- `constructor`: to initialize the contract with the block number of the execution of re balance.
+
+#### State Changing Functions
+
+The smart contract will have the following state changing functions:
+
+- `registerRetired`: to register retired details. Retired stores the retired address and approvers
```

Comment:
@aidan-kwon @toniya-klaytn 
Do we have to forbid the permission of registering retired addresses? what if a crazy guy executes the transaction with unknown address?

Reply:
yes this function is restricted to onlyOwner and the `approve` function verifies the msg.sender. So crazy guy cannot execute the transaction.